notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+   "outputs": [],
+   "source": [
+    "sql = '''\n",
+    "REPLACE INTO \"kttm_transformation\" OVERWRITE ALL\n",
+    "WITH \"ext\" AS \n",
+    "(\n",
+    "    SELECT *\n",
+    "    FROM TABLE(\n",
+    "      EXTERN(\n",
+    "        '{\"type\":\"http\",\"uris\":[\"https://static.imply.io/example-data/kttm-nested-v2/kttm-nested-v2-2019-08-25.json.gz\"]}',\n",
+    "        '{\"type\":\"json\"}'\n",
+    "      )\n",
+    "    ) EXTEND (\"timestamp\" VARCHAR, \"session\" VARCHAR, \"number\" VARCHAR, \"event\" TYPE('COMPLEX<json>'), \"agent\" TYPE('COMPLEX<json>'), \"client_ip\" VARCHAR, \"geo_ip\" TYPE('COMPLEX<json>'), \"language\" VARCHAR, \"adblock_list\" VARCHAR, \"app_version\" VARCHAR, \"path\" VARCHAR, \"loaded_image\" VARCHAR, \"referrer\" VARCHAR, \"referrer_host\" VARCHAR, \"server_ip\" VARCHAR, \"screen\" VARCHAR, \"window\" VARCHAR, \"session_length\" BIGINT, \"timezone\" VARCHAR, \"timezone_offset\" VARCHAR)\n",
+    ")\n",
+    "SELECT\n",
+    "  session, \n",
+    "  number,\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  TIMESTAMPDIFF(DAY, TIME_FLOOR(TIME_PARSE(\"timestamp\"), 'P1W'), TIME_PARSE(\"timestamp\")) AS days_since_week_start,\n",
+    "  TIME_FLOOR(TIME_PARSE(\"timestamp\"), 'P1W') AS week_start,\n",
+    "  TIME_CEIL(TIME_PARSE(\"timestamp\"), 'P1W') AS week_end,\n",
+    "  TIME_SHIFT(TIME_FLOOR(TIME_PARSE(\"timestamp\"), 'P1D'),'P1D', -1) AS start_of_yesterday,\n",
+    "  \n",
+    "  JSON_VALUE(\"event\", '$.percentage' RETURNING BIGINT) as percent_cleared,\n",
+    "  JSON_VALUE(\"geo_ip\", '$.city') AS city,\n",
+    "  \n",
+    "  CASE WHEN UPPER(\"adblock_list\")='NOADBLOCK' THEN 0 ELSE 1 END AS adblock_count,\n",
+    "  CASE WHEN UPPER(\"adblock_list\")='EASYLIST' THEN 1 ELSE 0 END AS easylist_count,\n",
+    "  \n",
+    "  REPLACE(REGEXP_EXTRACT(\"app_version\", '[^\\.]*\\.'),'.','') AS major_version,\n",
+    "  ARRAY_ORDINAL(STRING_TO_ARRAY(\"app_version\",'\\.'),2) AS minor_version,\n",
+    "  ARRAY_ORDINAL(STRING_TO_ARRAY(\"app_version\",'\\.'),3) AS patch_version,\n",
+    "  session_length\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "druid.display.run_task(sql)\n",
+    "druid.sql.wait_until_ready('kttm_transformation')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "40e787fe-65f5-4110-bc69-ce583285ba41",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# let's see what time of day shows the highest user activity\n",
+    "druid.display.sql(\"\"\"\n",
+    "SELECT EXTRACT( HOUR FROM \"__time\") time_hour, city, count(distinct \"session\") session_count\n",
+    "FROM \"kttm_transformation\" \n",
+    "WHERE \"city\" IS NOT NULL\n",
+    "GROUP BY 1,2 \n",
+    "ORDER BY 3 DESC \n",
+    "LIMIT 10\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9f3d47d9-ceaa-4623-9e8e-215b3ef391a4",
+   "metadata": {},
+   "source": [
+    "#### Nested Columns\n",
```

Comment:
All the below needs to go into the nested columns notebook.

Reply:
ok

Reply:
separate task though